test/e2e/gardener/shoot/create_hibernate_wakeup_delete.go

```diff
@@ -15,118 +14,137 @@ import (
 	"k8s.io/apimachinery/pkg/runtime/serializer"
 	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
 	"k8s.io/utils/ptr"
-	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 
 	"github.com/gardener/gardener/extensions/pkg/util"
 	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
 	v1beta1helper "github.com/gardener/gardener/pkg/apis/core/v1beta1/helper"
 	api "github.com/gardener/gardener/pkg/provider-local/apis/local"
 	"github.com/gardener/gardener/pkg/provider-local/apis/local/install"
 	. "github.com/gardener/gardener/pkg/utils/test/matchers"
-	e2e "github.com/gardener/gardener/test/e2e/gardener"
-	"github.com/gardener/gardener/test/e2e/gardener/shoot/internal/inclusterclient"
+	. "github.com/gardener/gardener/test/e2e/gardener"
+	. "github.com/gardener/gardener/test/e2e/gardener/shoot/internal"
 	"github.com/gardener/gardener/test/e2e/gardener/shoot/internal/node"
 )
 
 var _ = Describe("Shoot Tests", Label("Shoot", "default"), func() {
-	test := func(shoot *gardencorev1beta1.Shoot) {
-		f := defaultShootCreationFramework()
-		f.Shoot = shoot
+	Describe("Create, Hibernate, Wake up and Delete Shoot", func() {
+		test := func(s *ShootContext) {
+			if s.Shoot.Spec.CloudProfileName == nil && s.Shoot.Spec.CloudProfile != nil && s.Shoot.Spec.CloudProfile.Kind == "NamespacedCloudProfile" {
+				originalNamespacedCloudProfile := DefaultNamespacedCloudProfile()
+				namespacedCloudProfile := addCustomMachineImage(originalNamespacedCloudProfile.DeepCopy())
 
-		It("Create, Hibernate, Wake up and Delete Shoot", Offset(1), func() {
-			ctx, cancel := context.WithTimeout(parentCtx, 30*time.Minute)
-			defer cancel()
+				BeforeAll(func() {
+					DeferCleanup(func(ctx SpecContext) {
+						Eventually(func(g Gomega) {
```

Comment:
Can you ensure to always pass the context to `Eventually`?
```suggestion
						Eventually(ctx, func(g Gomega) {
```